### PR TITLE
#97 - Validate anonymizers options

### PR DESCRIPTION
**Pull request by SimonMellerin**

### CODE REVIEW

File: src/Anonymization/Anonymizer/AbstractAnonymizer.php [hunk 1]
```diff
@@ -99,6 +99,21 @@ protected function getSalt(): string
...
+     * an option `foo` is given with the correct type.
+     *
+     * This method is launch before the beginning of the anonymization process.
+     *
+     * If you override this method, you must call `parent::validateOptions()`
+     * at the begining of your implementation.
+     *
+     * @throws \Exception if given options are not valide.
+     */
+    public function validateOptions(): void {}
```
Comment (Lonnytunes) [inline src/Anonymization/Anonymizer/AbstractAnonymizer.php:115]:
```suggestion
    public function validateOptions(): void
    {
    }
```

File: src/Anonymization/Anonymizer/AbstractAnonymizer.php [hunk 2]
```diff
@@ -99,6 +99,21 @@ protected function getSalt(): string
         return $this->options->get('salt') ?? Anonymizator::generateRandomSalt();
     }
 
+    /**
+     * Validate Options given to the Anonymizer.
```
Comment (Lonnytunes) [inline src/Anonymization/Anonymizer/AbstractAnonymizer.php:103]:
```suggestion
     * Validate options given to the Anonymizer.
```

File: src/Anonymization/Anonymizer/AbstractAnonymizer.php (hunk 1 above)
Comment (pounard) [reply src/Anonymization/Anonymizer/AbstractAnonymizer.php:115]: CS fixer will not be happy, it puts empty braces in a single line.

Comment (Lonnytunes) [reply src/Anonymization/Anonymizer/AbstractAnonymizer.php:115]: Ah! I thought it was the opposite. My suggestion aimed to stick PHP CS fixer rules :sweat_smile:

File: src/Anonymization/Anonymizer/AbstractAnonymizer.php [hunk 3]
```diff
@@ -99,6 +99,21 @@ protected function getSalt(): string
...
+     *
+     * Override this method for your needs, for example validate that
+     * an option `foo` is given with the correct type.
+     *
+     * This method is launch before the beginning of the anonymization process.
+     *
+     * If you override this method, you must call `parent::validateOptions()`
+     * at the begining of your implementation.
```
Comment (Lonnytunes) [inline src/Anonymization/Anonymizer/AbstractAnonymizer.php:111]:
```suggestion
     * at the beginning of your implementation.
```

File: src/Anonymization/Anonymizer/AbstractAnonymizer.php [hunk 4]
```diff
@@ -99,6 +99,21 @@ protected function getSalt(): string
...
+     *
+     * @throws \Exception if given options are not valide.
```
Comment (Lonnytunes) [inline src/Anonymization/Anonymizer/AbstractAnonymizer.php:113]:
```suggestion
     * @throws \Exception if any option is invalid.
```

**Review by Lonnytunes — APPROVED** [review]

**Pull request merged**